src/compiler/types.ts

```diff
@@ -3532,6 +3532,20 @@ namespace ts {
         hasNoDefaultLib: boolean;
 
         languageVersion: ScriptTarget;
+
+        /**
+         * When `module` is `Node12` or `NodeNext`, this field controls weather the
+         * source file in question is an ESNext-output-format file, or a CommonJS-output-format
+         * module. This is derived by the module resolver as it looks up the file, since
+         * it is derived from either the file extension of the module, or the containing
+         * `package.json` context, and affects both checking and emit.
+         *
+         * It is _public_ so that (pre)transformers can set this field,
+         * since it switches the builtin `node` module transform. Generally speaking, if unset,
+         * the field is treated as though it is `ModuleKind.CommonJS`.
+         */
+        impliedNodeFormat?: ModuleKind.ESNext | ModuleKind.CommonJS;

```

Comment:
is there a `nodeFormat` property already? Would it be possible to call this `nodeFormat`? Does that name imply more certainty than this one?

Reply:
no.... `nodeFormat` just sounds like it might have something to do with `ts.Node` (since it's a property on one), while the extra word `implied` I'm hoping makes it more at-a-glance apparent that it has something to do with node-the-platform.